firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/FirebaseCrashlyticsReportManager.java

```diff
@@ -75,6 +80,11 @@ public void onNonFatalEvent(Throwable event, Thread thread) {
     onEvent(event, thread, EVENT_TYPE_LOGGED, false);
   }
 
+  @Override
+  public void onLog(long timestamp, String log) {
```

Comment:
This makes sense, but I want to talk out a naming thing that is going on here: You designed this manager as a "listener" as events. `onBeginSession`, `onFatalEvent`, `onEndSession` all make sense to me as lifecycle events because they're outside of the operation of this collaborator, and crashlytics is acting as *listener* of them. `onLog` (and to some degree,`onNonFatalEvent`) are a little different because the user explicitly calls it, right? In this case, we're not listening to a log occuring, we're listening to the app logging the information. I find it a little weird that `onLog` results in *logging*, but 🤷‍♂ I can't really think of a different way to name this.

Reply:
In my mind, the "event" happening here is a "user requests a log to be written" event, which is one event that can happen during the lifecycle of a session. (I'm pretty certain I'm going to rename the interface to `SessionLifecycleEvents`, as it happens).

Similar to a "button press" event, it's user-initiated, but still an event.

Reply:
It isn't the end user requesting that log though, it's likely some internal collaborator class. If you were implementing the code for onButtonPress, you would probably have code in a controller that called a class with a method that looked like `sendForm(...)`, not one that looked like `onFormSent(...)` The collaborator is requesting the sendForm action being taken, in that example. The expectation is a form gets sent.

I think the lifecycle idea works for session start / session ends, because you're abstracting away the meaning of them, but if we're planning to add lots of methods like onCustomKeys(...) it makes me wonder how much the lifecycle naming works.

Reply:
Talked about this offline, decided there's a good opportunity for some better interfacing here, but we'll break that out in a separate PR.